pkg/k8s/deployments/crud_test.go

```diff
@@ -208,9 +208,6 @@ func TestGet(t *testing.T) {
 			if tt.expectedFoundCount > 0 && d == nil {
 				t.Fatalf("expected %d deployments, instead found none", tt.expectedFoundCount)
 			}
-			if tt.expectedFoundCount == 0 && d != nil {
```

Comment:
I removed this because it the new versions of client-go, the fake client returns empty resources instead of nil when the resource doesn't exit (along with the error). https://github.com/kubernetes/client-go/commit/3d4f98db0e40e6bf0a0d6e48d4ef2e37365f6ba3#diff-5f29a60c4fc9f826637dd6d5496770ec8a20b960dc6b23433fc7c312a0351f0bR121

So, as we are already checking the error, and verifying is a not found when a deployment is not present, I think this can be removed without affecting the test